development/mob_programming.md

```diff
@@ -0,0 +1,45 @@
+
+If you do it formally, there are rules:
+
+* One person is the _driver_, and they don't make any decisions about the code or configuration that's being written
+* Another may be nominated as the _navigator_, primarily making the suggestions, or feedback may come from the whole group
+* The driver rotates in quite short periods, 5 to 10 minutes
+
+## Helping it work for research
+
+The formal rules don't always work well in a research setting where there are a lot of speculative decisions to be made, different paths to go a short way down, and often complex dependencies. Here are some rules of thumb developed in different settings of mob programming with a mixed group of software engineers and researchers.
+
+* Keep the group size to no more than 4, ideally 3 - larger than this, and people tend to drift off.
+* Try to rotate the screenshare at least every 20 minutes to help keep everyone engaged, though there may be some settings where it makes sense for the least experienced person to "drive" the whole way through
+* When starting a new project or onboarding a new person, allow at least a whole session for environment setup
+* Sessions work well at a length of 1.5-2 hours, including a screenbreak; screenbreaks are important!
+* The practise works well remotely, probably better than it does in-person (opinions may vary)
+* Limit time spent mobbing to half a working day (typically 10:30-12:00, 2:00-4:00) or it can get cognitively exhausting; plan for quiet days
+* Articulate shared intentions at the start of the session - where are we aiming to be at the end of it? It helps to take a note for the benefit of people who can't be there.
+* Mob sessions are "best effort" and no-one should feel obliged to participate in them
+* It's ok to spend whole sessions just reading code or papers together. 
+* Optimise mob sessions for work _where decisions need to be made_ - if a path of development is very clearly laid out, leave that available for team members who need to work asynchronously (e.g. they have caring commitments, or very different patterns of brain-waking times)
+
+The [Recurse Center social rules](https://www.recurse.com/social-rules) apply; kindness is key!
+
+## What to watch out for 
+
+Experience also says that starting to mob program with a new team can be a painful experience, what can happen is 
+
+* Really experienced contributors can feel hobbled
+* People without much background can feel a bit lost
+* Delivery can feel _much_ slower than hammering something out on one's own
```

Comment:
Do all 3 of these point related to the paragraphs below?  It feels like they do, but perhaps the order is not the same.   I feel like the paragraphs should directly relate, then you have potential problem with reasoning and a solution to mitigate?

Reply:
I appreciate the attention to detail but this feels like over-reading - as commented above, this isn't a sales pitch, rather a "what to watch out for" and the solution will vary according to your experiences :D